source_gen/CHANGELOG.md

```diff
@@ -8,6 +8,9 @@
   * These changes will likely have no effect on output, unless you customize
     the code formatter.
 
+* Introduce `SharedPartBuilder` for creating part files that can be merged
```

Comment:
I think probably we'll want to make a breaking change in the existing `PartBuilder` to make the extension argument required. I think we probably want everything to go out together. Should we drop a release of the current stuff before adding this?